forte/data/readers/openie_reader.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2019 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+The reader that reads Open-IE extractions data into data pack.
+Format:
+https://github.com/gabrielStanovsky/oie-benchmark/tree/master/oie_corpus
+"""
+import logging
+import os
+from typing import Iterator, Any
+
+from forte.common.exception import ProcessorConfigError
+from forte.common.configuration import Config
+from forte.common.resources import Resources
+from forte.data.data_utils_io import dataset_path_iterator
+from forte.data.data_pack import DataPack
+from forte.data.readers.base_reader import PackReader
+from ft.onto.base_ontology import Sentence, PredicateMention, \
+    PredicateArgument, PredicateLink, Document
+
+__all__ = [
+    "OpenIEReader"
+]
+
+
+class OpenIEReader(PackReader):
+    r""":class:`OpenIEReader` is designed to read in the Open IE dataset used
+        by Open Information Extraction task. The related paper can be found
+        `here
+        <https://gabrielstanovsky.github.io/assets/papers/emnlp16a/paper.pdf>`__.
+        The related source code for generating this dataset can be found
+        `here
+        <https://github.com/gabrielStanovsky/oie-benchmark>`__.
+        To use this Reader, you must follow the dataset format described
+        `here
+        <https://github.com/gabrielStanovsky/oie-benchmark/tree/master/oie_corpus>`__.
+    """
+
+    def _collect(self, *args, **kwargs) -> Iterator[Any]:
+        # pylint: disable = unused-argument
+        r"""Should be called with param ``oie_directory`` which is a path to a
+        folder containing json files.
+
+        Args:
+            args: args[0] is the directory to the open ie files.
+            kwargs:
+
+        Returns: Iterator over files in the path with oie extensions.
+        """
+        oie_directory = args[0]
+        oie_file_extension = self.configs.oie_file_extension
+        logging.info("Reading dataset from %s with extension %s",
+                     oie_directory, oie_file_extension)
+        return dataset_path_iterator(oie_directory, oie_file_extension)
+
+    def _cache_key_function(self, oie_file: str) -> str:
+        return os.path.basename(oie_file)
+
+    def _parse_pack(self, file_path: str) -> Iterator[DataPack]:
+        pack = self.new_pack()
+        with open(file_path, "r", encoding="utf8") as f:
```

Comment:
Oh, I just realize you should better locate the code that depends on the `doc` inside the `with` block. Also you don't have to read all the lines first to save memeory.


```suggestion
        with open(file_path, "r", encoding="utf8") as f:
               for line in doc:
                    line = line.strip()
```

Reply:
Done. Now the parsing is within the `with` block and processed line by line instead of reading all lines ahead.